### PR TITLE
Enable pandas pytests for `cudf.pandas`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
This PR enables `cudf.pandas` pandas pytest suite.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -155,17 +155,17 @@ jobs:
...
-  #     script: ci/cudf_pandas_scripts/pandas-tests/run.sh pr
-  #     # Hide test failures because they exceed the GITHUB_STEP_SUMMARY output limit.
-  #     test_summary_show: "none"
+  pandas-tests:
+    # run the Pandas unit tests using PR branch
+    needs: wheel-build-cudf
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@branch-24.04
+    with:
+      matrix_filter: map(select(.ARCH == "amd64")) | max_by(.CUDA_VER) | [.]
```
Comment (galipremsagar) [inline .github/workflows/pr.yaml:164]:
```suggestion
      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.10")) | max_by(.CUDA_VER) | [.]
```

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -114,15 +114,15 @@ jobs:
...
-  #     date: ${{ inputs.date }}
-  #     sha: ${{ inputs.sha }}
-  #     # pr mode uses the HEAD of the branch, which is also correct for nightlies
-  #     script: ci/cudf_pandas_scripts/pandas-tests/run.sh pr
+  pandas-tests:
+    # run the Pandas unit tests
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@branch-24.04
+    with:
+      matrix_filter: map(select(.ARCH == "amd64")) | max_by(.CUDA_VER) | [.]
```
Comment (galipremsagar) [inline .github/workflows/test.yaml:122]:
```suggestion
      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.10")) | max_by(.CUDA_VER) | [.]
```

File: .github/workflows/pr.yaml [hunk 3]
```diff
@@ -155,17 +155,17 @@ jobs:
...
-  #     script: ci/cudf_pandas_scripts/pandas-tests/run.sh pr
-  #     # Hide test failures because they exceed the GITHUB_STEP_SUMMARY output limit.
-  #     test_summary_show: "none"
+  pandas-tests:
+    # run the Pandas unit tests using PR branch
+    needs: wheel-build-cudf
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@branch-24.04
+    with:
+      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.10")) | max_by(.CUDA_VER) | [.]
```
Comment (galipremsagar) [inline .github/workflows/pr.yaml:164]:
```suggestion
      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.9")) | max_by(.CUDA_VER) | [.]
```

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -114,15 +114,15 @@ jobs:
...
-  #     date: ${{ inputs.date }}
-  #     sha: ${{ inputs.sha }}
-  #     # pr mode uses the HEAD of the branch, which is also correct for nightlies
-  #     script: ci/cudf_pandas_scripts/pandas-tests/run.sh pr
+  pandas-tests:
+    # run the Pandas unit tests
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@branch-24.04
+    with:
+      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.10")) | max_by(.CUDA_VER) | [.]
```
Comment (galipremsagar) [inline .github/workflows/test.yaml:122]:
```suggestion
      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.9")) | max_by(.CUDA_VER) | [.]
```

File: .github/workflows/test.yaml [hunk 5]
```diff
@@ -114,15 +114,15 @@ jobs:
...
-  #     date: ${{ inputs.date }}
-  #     sha: ${{ inputs.sha }}
-  #     # pr mode uses the HEAD of the branch, which is also correct for nightlies
-  #     script: ci/cudf_pandas_scripts/pandas-tests/run.sh pr
+  pandas-tests:
+    # run the Pandas unit tests
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@branch-24.04
+    with:
+      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.9")) | max_by(.CUDA_VER) | [.]
```
Comment (galipremsagar) [inline .github/workflows/test.yaml:122]:
```suggestion
      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.11")) | max_by(.CUDA_VER) | [.]
```

File: .github/workflows/pr.yaml [hunk 6]
```diff
@@ -155,17 +155,17 @@ jobs:
...
-  #     script: ci/cudf_pandas_scripts/pandas-tests/run.sh pr
-  #     # Hide test failures because they exceed the GITHUB_STEP_SUMMARY output limit.
-  #     test_summary_show: "none"
+  pandas-tests:
+    # run the Pandas unit tests using PR branch
+    needs: wheel-build-cudf
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@branch-24.04
+    with:
+      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.9")) | max_by(.CUDA_VER) | [.]
```
Comment (galipremsagar) [inline .github/workflows/pr.yaml:164]:
```suggestion
      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.11")) | max_by(.CUDA_VER) | [.]
```

**Review by AyodeAwe — APPROVED** [review]

File: python/cudf/cudf/pandas/scripts/run-pandas-tests.sh [hunk 7]
```diff
@@ -22,7 +22,7 @@ set -euo pipefail
 # of Pandas installed.
 PANDAS_VERSION=$(python -c "import pandas; print(pandas.__version__)")
 
-PYTEST_IGNORES="--ignore=tests/io/test_user_agent.py --ignore=tests/interchange/test_impl.py --ignore=tests/window/test_dtypes.py --ignore=tests/strings/test_api.py --ignore=tests/window/test_numba.py"
+PYTEST_IGNORES="--ignore=tests/io/test_user_agent.py --ignore=tests/interchange/test_impl.py --ignore=tests/window/test_dtypes.py --ignore=tests/strings/test_api.py --ignore=tests/window/test_numba.py --ignore=tests/window --ignore=tests/io/pytables --ignore=tests/plotting --ignore=tests/scalar --ignore=tests/series/test_arithmetic.py --ignore=tests/tslibs/test_parsing.py --ignore=tests/io/parser/common/test_read_errors.py"
```
[the remark this answers is not in the document]
Comment (galipremsagar) [reply python/cudf/cudf/pandas/scripts/run-pandas-tests.sh]: Done 👍

File: ci/cudf_pandas_scripts/pandas-tests/run.sh [hunk 8]
```diff
@@ -37,6 +37,7 @@ bash python/cudf/cudf/pandas/scripts/run-pandas-tests.sh \
   -m "not slow" \
```
Comment (wence-) [inline ci/cudf_pandas_scripts/pandas-tests/run.sh:37]: *suggestion*:

I found that `--tb=no` is much faster (and produces less output) than the other options.

**Review by wence- — COMMENTED** [review]
So the idea is that any tests that fail are non-fatal (and don't affect the overall workflow outcome), but any "other" failures (setup/whatever) _are_.

Do we have any observability over the level of pass/fail other than explicitly looking in the job log?

File: ci/cudf_pandas_scripts/pandas-tests/run.sh (hunk 8 above)
Comment (galipremsagar) [reply ci/cudf_pandas_scripts/pandas-tests/run.sh:37]: Done

**Comment by galipremsagar** [comment]
> So the idea is that any tests that fail are non-fatal (and don't affect the overall workflow outcome), but any "other" failures (setup/whatever) _are_.

Yes, but I'm planning on adding the diff calculator for each of the test run in follow-up PRs. This PR just enables what we had disabled couple of weeks ago.

> Do we have any observability over the level of pass/fail other than explicitly looking in the job log?

Right now, we have to look at test summary. But as I said above, after this PR merges I'll be enabling the diff calculation and commenting(if possible).

**Comment by wence-** [comment]
> > So the idea is that any tests that fail are non-fatal (and don't affect the overall workflow outcome), but any "other" failures (setup/whatever) _are_.
> 
> Yes, but I'm planning on adding the diff calculator for each of the test run in follow-up PRs. This PR just enables what we had disabled couple of weeks ago.
> 
> > Do we have any observability over the level of pass/fail other than explicitly looking in the job log?
> 
> Right now, we have to look at test summary. But as I said above, after this PR merges I'll be enabling the diff calculation and commenting(if possible).

Thanks!

**Review by wence- — APPROVED** [review]

**Pull request merged**